crates/accelerate/src/sabre_swap/mod.rs

```diff
@@ -621,103 +632,102 @@ fn route_reachable_nodes<F>(
 /// routable in terms of number of swaps.
-fn closest_operation(
-    front_layer: &FrontLayer,
-    layout: &NLayout,
-    dist: &ArrayView2<f64>,
-) -> (NodeIndex, [VirtualQubit; 2]) {
-    let (&node, qubits) = front_layer
+fn closest_operation<'a>(
+    front_layer: &'a FrontLayer,
+    dist: &'_ ArrayView2<f64>,
+) -> (&'a NodeIndex, &'a [PhysicalQubit; 2]) {
+    front_layer
         .iter()
-        .map(|(node, qubits)| (node, [qubits[0].to_phys(layout), qubits[1].to_phys(layout)]))
         .min_by(|(_, qubits_a), (_, qubits_b)| {
             dist[[qubits_a[0].index(), qubits_a[1].index()]]
                 .partial_cmp(&dist[[qubits_b[0].index(), qubits_b[1].index()]])
                 .unwrap_or(Ordering::Equal)
         })
-        .unwrap();
-    (node, [qubits[0].to_virt(layout), qubits[1].to_virt(layout)])
+        .unwrap()
 }
 
 /// Add the minimal set of swaps to the `swaps` vector that bring the two `qubits` together so that
 /// a 2q gate on them could be routed.
 fn swaps_to_route(
-    swaps: &mut Vec<[VirtualQubit; 2]>,
-    qubits: &[VirtualQubit; 2],
-    layout: &NLayout,
+    swaps: &mut Vec<[PhysicalQubit; 2]>,
+    qubits: &[PhysicalQubit; 2],
     coupling_graph: &DiGraph<(), ()>,
 ) {
     let mut shortest_paths: DictMap<NodeIndex, Vec<NodeIndex>> = DictMap::new();
     (dijkstra(
         coupling_graph,
-        NodeIndex::new(qubits[0].to_phys(layout).index()),
-        Some(NodeIndex::new(qubits[1].to_phys(layout).index())),
+        NodeIndex::new(qubits[0].index()),
+        Some(NodeIndex::new(qubits[1].index())),
         |_| Ok(1.),
         Some(&mut shortest_paths),
     ) as PyResult<Vec<Option<f64>>>)
         .unwrap();
     let shortest_path = shortest_paths
-        .get(&NodeIndex::new(qubits[1].to_phys(layout).index()))
+        .get(&NodeIndex::new(qubits[1].index()))
         .unwrap()
         .iter()
         .map(|n| PhysicalQubit::new(n.index() as u32))
         .collect::<Vec<_>>();
-    // Insert greedy swaps along that shortest path
+    // Insert greedy swaps along that shortest path, splitting them between moving the left side
+    // and moving the right side to minimise the depth.  One side needs to move up to the split
+    // point and the other can stop one short because the gate will be routable then.
     let split: usize = shortest_path.len() / 2;
-    let forwards = &shortest_path[1..split];
-    let backwards = &shortest_path[split..shortest_path.len() - 1];
     swaps.reserve(shortest_path.len() - 2);
-    for swap in forwards {
-        swaps.push([qubits[0], swap.to_virt(layout)]);
+    for i in 0..split {
+        swaps.push([shortest_path[i], shortest_path[i + 1]]);
     }
-    for swap in backwards.iter().rev() {
-        swaps.push([qubits[1], swap.to_virt(layout)]);
+    for i in 0..split - 1 {
+        let end = shortest_path.len() - 1 - i;
+        swaps.push([shortest_path[end], shortest_path[end - 1]]);
```

Comment:
Just for the record in case we're looking at this again in the future. This took me a while to trace through in my head before I realized these were equivalent. The tricky bit I missed at first when comparing the old to the new is the difference between the virtual and physical qubits, the old path worked with virtual qubits and the new one is using physical. This is somewhere having #10761 is important to validate we're working in the correct domain because if they were just integers this would have been easy to mess up.